src/Control/Monad/Trans/Instances.hs

```diff
@@ -1248,3 +1252,129 @@ data C1_0'SelectT
 #  endif
 # endif
 #endif
+
+#if !(MIN_VERSION_base(4,9,0))
+#if MIN_VERSION_transformers(0,4,0) && !(MIN_VERSION_transformers(0,5,0))
+
+#if MIN_VERSION_base(4,4,0)
+instance Eq1 Complex where eq1 = (==)
+instance Read1 Complex where readsPrec1 = readsPrec
+instance Show1 Complex where showsPrec1 = showsPrec
+#endif
+
+instance (Eq a, Eq b) => Eq1 ((,,) a b) where eq1 = (==)
+instance (Ord a, Ord b) => Ord1 ((,,) a b) where compare1 = compare
+instance (Read a, Read b) => Read1 ((,,) a b) where readsPrec1 = readsPrec
+instance (Show a, Show b) => Show1 ((,,) a b) where showsPrec1 = showsPrec
+
+instance (Eq a, Eq b, Eq c) => Eq1 ((,,,) a b c) where eq1 = (==)
+instance (Ord a, Ord b, Ord c) => Ord1 ((,,,) a b c) where compare1 = compare
+instance (Read a, Read b, Read c) => Read1 ((,,,) a b c) where readsPrec1 = readsPrec
+instance (Show a, Show b, Show c) => Show1 ((,,,) a b c) where showsPrec1 = showsPrec
+#else
```

Comment:
Everything after this `#else` will be reachable if `TRANSFORMERS_FOUR` is not defined. This technically works, but it's a little unsatisfactory. This is because when `transformers-compat` is built with the `-ftwo` or `-fthree` flags, it backports the `Data.Functor.Classes` in its own module [here](https://github.com/ekmett/transformers-compat/blob/b7e02707590bd7b9da766babe7886b01cbca9306/0.3/Data/Functor/Classes.hs). I've made an attempt to avoid orphan instances whenever possible by defining the appropriate instances in that module. That way, if you import `Data.Functor.Classes`, you'll only need to import `Control.Monad.Trans.Instances ()` to get orphan instances if you're using a recent version of `transformers`.

Following the example set by [this code](https://github.com/ekmett/transformers-compat/blob/b7e02707590bd7b9da766babe7886b01cbca9306/src/Control/Monad/Trans/Instances.hs#L442-L471) in `Control.Monad.Trans.Instances` and its [corresponding code in `0.3/Data/Functor/Classes.hs`](https://github.com/ekmett/transformers-compat/blob/b7e02707590bd7b9da766babe7886b01cbca9306/0.3/Data/Functor/Classes.hs#L397-L410), can you:

1. Change this `#else` to:

   ```suggestion
   # elif MIN_VERSION_transformers(0,5,0)
   ```
2. Copy the code below to `0.3/Data/Functor/Classes.hs` as well.